### PR TITLE
Logging tweaks and fixes

### PR DESCRIPTION
**Pull request by optimumtact**
Added a new log file for viruses

Logs each infect event and contains detailed descriptions of the virus

I also added a log message for printing virus culture bottles including
who printed them, this goes to the same log file

I added a game log entry for facehugger impregnation
because why not

Medbots now log their injections and reagents in the combat log

combat log helper no longer tries to run key_name on passed in objects

Ghost/Dead say no longer logs twice needlessly, as the log entry from mob
observer dead is now removed as it calls say_dead anyway

Possible todos:
I might add some more logs around the method of infection and I'll also
see if I can find exact logs for virus culture bottle infections, since we want
to track those in more detail I suspect.

### CODE REVIEW

**Comment by optimumtact** [comment]
PS it looks like this
```
:~/tgstation/data/logs/2019/02/17/round-09.43.25$ tail -f virus.                                                                                                                            log
[2019-02-17 09:45:25.427] VIRUS: Optimumtact/(Gabriela Leichter) was infected by                                                                                                                             virus: The Flu : /datum/disease/flu at (Captain's Office (120, 125, 2))
[2019-02-17 09:46:10.027] VIRUS: A culture bottle was printed for the virus Cold                                                                                                                             sym:Sneezing r:3 s:-2 ss:0 t:4 at (Virology (155, 95, 2)) by Optimumtact/(Gabri                                                                                                                            ela Leichter)
[2019-02-17 09:47:40.786] VIRUS: A culture bottle was printed for the virus Unkn                                                                                                                            own sym:Sneezing and Viral evolutionary acceleration r:0 s:-4 ss:5 t:7 at (Virol                                                                                                                            ogy (155, 95, 2)) by Optimumtact/(Gabriela Leichter)
```

**Comment by optimumtact** [comment]
```
[2019-02-17 10:37:10.167] ATTACK: *no key*/(�Medibot) has injected Optimumtact/(Gabriela Leichter) with beaker source Unholy Water, 50:15 (NEWHP: 75.5)  (Central Primary Hallway (129, 117, 2))
[2019-02-17 10:37:16.167] ATTACK: *no key*/(�Medibot) has injected Optimumtact/(Gabriela Leichter) with internal synthesizer bicaridine:15 (NEWHP: 64.5)  (Central Primary Hallway (129, 117, 2))
[2019-02-17 10:37:22.167] ATTACK: *no key*/(�Medibot) has injected Optimumtact/(Gabriela Leichter) with internal synthesizer kelotane:15 (NEWHP: 53.3)  (Central Primary Hallway (129, 117, 2))
```

**Comment by Cyprex** [comment]
Looks great! Are APC interactions currently logged?

File: code/__DEFINES/logging.dm [hunk 1]
```diff
@@ -18,22 +18,23 @@
...
-#define LOG_DSAY		(1 << 4)
-#define LOG_PDA			(1 << 5)
-#define LOG_CHAT		(1 << 6)
-#define LOG_COMMENT		(1 << 7)
-#define LOG_TELECOMMS	(1 << 8)
-#define LOG_OOC			(1 << 9)
-#define LOG_ADMIN		(1 << 10)
-#define LOG_OWNERSHIP	(1 << 11)
-#define LOG_GAME		(1 << 12)
+#define LOG_ATTACK		  (1 << 0)
```
Comment (optimumtact) [inline code/__DEFINES/logging.dm]: I need to fix this

**Review by Tlaltecuhtli — CHANGES_REQUESTED** [review]

File: code/datums/diseases/advance/advance.dm [hunk 2]
```diff
@@ -117,6 +117,13 @@
 	//this is a new disease starting over at stage 1, so processing is not copied
 	return A
 
+//Describe this disease to an admin in detail (for logging)
+/datum/disease/advance/admin_details()
+	var/list/name_symptoms = list()
+	for(var/datum/symptom/S in symptoms)
+		name_symptoms += S.name
+	return "[name] sym:[english_list(name_symptoms)] r:[totalResistance()] s:[totalStealth()] ss:[totalStageSpeed()] t:[totalTransmittable()]"
```
[the remark this answers is not in the document]
Comment (Cyprex) [reply code/datums/diseases/advance/advance.dm]: I think this is fine.

**Comment by Cyprex** [comment]
#42806 is a recent logging issue, in case you are interested in fixing that here.
Having a central log for each AI (possibly in their PP) would be nice, as well as having logs for airlock interactions via AI interactions.

File: code/modules/mob/living/simple_animal/bot/medbot.dm [hunk 3]
```diff
@@ -490,15 +490,18 @@
...
 
 		var/failed = FALSE
-		if(do_mob(src, patient, 30))	//Is C == patient? This is so confusing
+		if(do_mob(src, patient, 30))
 			if((get_dist(src, patient) <= 1) && (on) && assess_patient(patient))
 				if(reagent_id == "internal_beaker")
 					if(use_beaker && reagent_glass && reagent_glass.reagents.total_volume)
 						var/fraction = min(injection_amount/reagent_glass.reagents.total_volume, 1)
+						var/reagentlist = pretty_string_from_reagent_list(reagent_glass.reagents.reagent_list)
+						log_combat(src, patient, "injected", "beaker source", "[reagentlist]:[injection_amount]")
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/bot/medbot.dm]: `"beaker source"`
log_combat expects an atom for that param. move it to the additional.

File: code/modules/mob/living/simple_animal/bot/medbot.dm [hunk 4]
```diff
@@ -490,15 +490,18 @@
...
 						reagent_glass.reagents.reaction(patient, INJECT, fraction)
 						reagent_glass.reagents.trans_to(patient,injection_amount) //Inject from beaker instead.
 				else
 					patient.reagents.add_reagent(reagent_id,injection_amount)
+					log_combat(src, patient, "injected", "internal synthesizer", "[reagent_id]:[injection_amount]")
```
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/bot/medbot.dm]: ditto

**Review by ShizCalev — DISMISSED** [review]

File: code/modules/mob/living/simple_animal/bot/medbot.dm (hunk 3 above)
Comment (optimumtact) [reply code/modules/mob/living/simple_animal/bot/medbot.dm]: it expects an atom, but doesn't actually need one

Comment (ShizCalev) [reply code/modules/mob/living/simple_animal/bot/medbot.dm]: have you tested it? iirc this results in the message just saying "no key"

https://github.com/tgstation/tgstation/blob/49a4cb67694a1f36c36fa016c9686721d11b4075/code/game/atoms.dm#L688
https://github.com/tgstation/tgstation/blob/49a4cb67694a1f36c36fa016c9686721d11b4075/code/game/atoms.dm#L696-L697

https://github.com/tgstation/tgstation/blob/eac76e84eae260c85c7c99e17e70964ad3a96711/code/__HELPERS/_logging.dm#L253

Comment (optimumtact) [reply code/modules/mob/living/simple_animal/bot/medbot.dm]: I removed the key name because it was stupid, most things you are hit with are not mobs

Comment (optimumtact) [reply code/modules/mob/living/simple_animal/bot/medbot.dm]: and it was putting a [DC} on all objects

File: code/datums/diseases/advance/advance.dm (hunk 2 above)
[the remark this answers is not in the document]
Comment (subject217) [reply code/datums/diseases/advance/advance.dm]: using full names doubles log length, admins can learn things. the symptoms are the most important thing anyway.

you already need the guide to viro open to discover if the symptoms have their special abilities unlocked from the stats unless you are the arbiter of virology knowledge anyway lol

**Comment by optimumtact** [comment]
This is ready

File: code/__DEFINES/logging.dm (hunk 1 above)
Comment (Jordie0608) [inline code/__DEFINES/logging.dm]: No it ain't.

**Comment by Jordie0608** [comment]
Also remember the new log file will need to be added to the tgstation13.org parser list of files.

**Comment by optimumtact** [comment]
look at the raw and also, that's not my problem

**Comment by Jordie0608** [comment]
I assumed your comment on the defines was referring to the mix of tabs and spaces, which is still present, was that not correct?
capture

**Comment by optimumtact** [comment]
looks good to me

**Comment by Jordie0608** [comment]
Dumb berry sphere needs his glasses.

**Pull request merged**